apps/s_client.c

```diff
@@ -1370,6 +1376,20 @@ int s_client_main(int argc, char **argv)
         BIO_printf(bio_err, "%s: Can't use both -4 and -6\n", prog);
         goto opthelp;
     }
+    if (noservername) {
+        if (servername != NULL) {
+            BIO_printf(bio_err,
+                       "%s: Can't use -servername and -noservername together\n",
+                       prog);
+            goto opthelp;
+        }
+        if (dane_tlsa_domain != NULL) {
```

Comment:
If `dane_tlsa_domain` is set then it ends up setting the SNI name to that value (unless it is already set to something else). Therefore the `noservername` option is incompatible with this option.